### PR TITLE
build: bump uncrustify to version 0.79.0

### PR DESCRIPTION
**Pull request by clason**

### CODE REVIEW

File: src/uncrustify.cfg [hunk 1]
```diff
@@ -320,709 +317,713 @@ sp_byref_paren                  = ignore   # ignore/add/remove/force/not_defined
...
-sp_before_square_asm_block      = ignore   # ignore/add/remove/force/not_defined
+sp_before_square_asm_block      = ignore   # ignore/add/remove/force
 
 # Add or remove space before '[]'.
-sp_before_squares               = remove   # ignore/add/remove/force/not_defined
+sp_before_squares               = remove   # ignore/add/remove/force
+
+# Add or remove space before C++17 structured bindings
+# after byref.
+sp_cpp_before_struct_binding_after_byref = ignore   # ignore/add/remove/force
```
Comment (clason) [inline src/uncrustify.cfg:473]: new option (irrelevant)

File: src/uncrustify.cfg [hunk 2]
```diff
@@ -1095,34 +1096,37 @@ sp_num_before_emb_cmt           = 1        # unsigned number
...
 
 # Add or remove space after a bit colon ':'.
-sp_after_bit_colon              = ignore   # ignore/add/remove/force/not_defined
+sp_after_bit_colon              = ignore   # ignore/add/remove/force
 
 # If true, a <TAB> is inserted after #define.
 force_tab_after_define          = false    # true/false
 
+# Add or remove space between two strings.
+sp_string_string                = ignore   # ignore/add/remove/force
```
Comment (clason) [inline src/uncrustify.cfg:1128]: new option

File: src/uncrustify.cfg [hunk 3]
```diff
@@ -2103,41 +2107,45 @@ nl_oc_msg_args_min_params       = 0        # unsigned number
 # (OC) Max code width of Objective-C message before applying nl_oc_msg_args.
 nl_oc_msg_args_max_code_width   = 0        # unsigned number
 
+# (OC) Whether to apply nl_oc_msg_args if some of the parameters are already
+# on new lines. Overrides nl_oc_msg_args_min_params and nl_oc_msg_args_max_code_width.
+nl_oc_msg_args_finish_multi_line = false    # true/false
```
Comment (clason) [inline src/uncrustify.cfg:2112]: new option (what even is this?)

File: src/uncrustify.cfg [hunk 4]
```diff
@@ -76,238 +76,235 @@ utf8_force                      = false    # true/false
...
 
 # Add or remove space between nested parentheses, i.e. '((' vs. ') )'.
-sp_paren_paren                  = remove   # ignore/add/remove/force/not_defined
+sp_paren_paren                  = remove   # ignore/add/remove/force
 
 # Add or remove space between back-to-back parentheses, i.e. ')(' vs. ') ('.
-sp_cparen_oparen                = remove   # ignore/add/remove/force/not_defined
-
-# Whether to balance spaces inside nested parentheses.
-sp_balance_nested_parens        = false    # true/false
```
Comment (clason) [inline src/uncrustify.cfg:189]: > option 'sp_balance_nested_parens' never works; it has been replaced by 'sp_paren_paren'.

You don't say... 👀

File: src/uncrustify.cfg (hunk 2 above)
Comment (dundargoc) [reply src/uncrustify.cfg:1128]: Should be `force`

**Review by dundargoc — APPROVED** [review]

**Pull request merged**